plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/builders/utils/IpTool.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.orchestrator.plugin.huaweicloud.builders.utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import org.apache.commons.validator.routines.InetAddressValidator;
+
+/**
+ * A Tool to probe external IP.
+ */
+public class IpTool {
```

Comment:
The purpose of this class is not very clear to me. 
if it is to get external IP of the VM, then we will need some other better way of handling it. (Eg., getting it from the EIP resource)
Also we can make such methods as static methods. .

Reply:
`IpTool` is a tool class to probe the external IP of the host of the `xpanse` which will be used by the security-group-rules.
As cloud-init in the HuaweiCloud has some issues(still checking). I choose to use the `provisions` ability of terraform to implement the software installing.